libdrgn/dwarf_info.c

```diff
@@ -6097,6 +6097,17 @@ drgn_compound_type_from_dwarf(struct drgn_debug_info *dbinfo,
 			if (err)
 				goto err;
 			break;
+		case DW_TAG_GNU_template_parameter_pack:
+			r = dwarf_child(&child, &child_param);
+			while (r == 0) {
+				err = parse_template_parameter(dbinfo, module, &child_param,
```

Comment:
This assumes that the child is `DW_TAG_template_type_parameter`, but it may also be a `DW_TAG_template_value_parameter` (or neither if GCC/Clang decides to put some other information in there in the future).

So, this needs to check the child's tag and handle it just like the other places where we parse `DW_TAG_template_type_parameter`/`DW_TAG_template_value_parameter`. Since this would be the third copy of that logic (in addition to the one just above here and the one in `drgn_function_type_from_dwarf()`), it might be worth refactoring it about to get rid of the duplication, but I'll leave that up to you.

Reply:
Good point, fixed!